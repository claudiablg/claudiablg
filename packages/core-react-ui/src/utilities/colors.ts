import { Color } from '@newrade/core-design-system';

export function getHSLColor({ h, s, l, a }: Partial<Color> = { h: 200, s: 100, l: 100, a: 1 }) {
  return `hsla(${h}, ${s}%, ${l}%, ${a || 1})`;
}

export function lightenColor(color: Color, amount: number) {}

export function darkenColor(color: Color, amount: number) {}
