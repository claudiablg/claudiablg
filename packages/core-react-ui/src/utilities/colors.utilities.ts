import { Color } from '@newrade/core-design-system';

export function getHSLColor({ h, s, l, a }: Partial<Color>) {
  return `hsl(${h}deg ${s}% ${l}% / ${a || 100}%)`;
  // return `hsla(${h}, ${s}%, ${l}%, ${a || 1})`;
}

export function lightenColor(color: Color, amount: number) {}

export function darkenColor(color: Color, amount: number) {}
