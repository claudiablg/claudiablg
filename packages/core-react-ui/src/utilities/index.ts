export * from './colors.utilities';
export * from './component.utilities';
export * from './css-variable.utilities';
export * from './effects.utilities';
export * from './filter-color.utility';
export * from './text.utilities';
export * from './theme.utilities';
export * from './utilities';
