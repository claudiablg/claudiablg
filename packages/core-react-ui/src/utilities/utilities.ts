import { ColorIntents, Colors } from '@newrade/core-design-system';

/**
 * Create default color intents.
 */
export function createDefaultColorIntents(colors: Colors): ColorIntents {
  return {
    primary: colors.primary500,
    primaryReversed: colors.greyscale0,
    secondary: colors.primary500,
    secondaryReversed: colors.greyscale0,
    primaryText: colors.greyscale900,
    primaryTextReversed: colors.greyscale0,
    secondaryText: colors.greyscale700,
    secondaryTextReversed: colors.greyscale0,
    tertiaryText: colors.greyscale600,
    tertiaryTextReversed: colors.greyscale0,
    successText: colors.utilityGreen1000,
    successAction: colors.utilityGreen500,
    successBackground: colors.utilityGreen100,
    warningText: colors.utilityYellow1000,
    warningAction: colors.utilityYellow500,
    warningBackground: colors.utilityYellow100,
    dangerText: colors.utilityRed1000,
    dangerAction: colors.utilityRed500,
    dangerBackground: colors.utilityRed100,
    background0: colors.greyscale0,
    background1: colors.greyscale100,
    background2: colors.greyscale200,
  };
}

/**
 * Convert a number (in px) into a rem value.
 * @param px value in pixel
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxToRem({ baseUnitPx, value }: { baseUnitPx: number; value: number }): string {
  return `${value / baseUnitPx}rem`;
}

/**
 * Convert a number (in px) into a em relative value.
 * @param px value in pixel
 * @param fontSize font-size set on the element
 */
export function pxToEm({ fontSize, value }: { fontSize: number; value: number }): string {
  return `${value / fontSize}em`;
}

/**
 * Convert px value (e.g. '15px') into a rem value.
 * @param px value in string px value
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxStringToRem({ baseFontSize, value }: { baseFontSize: number; value: string }): string {
  if (!value?.length) {
    return '';
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return '';
  }
  const conversion = Number(match[0]);
  if (!conversion) {
    return '';
  }

  return `${conversion / baseFontSize}rem`;
}

/**
 * Convert px value (e.g. '15px') into a number value.
 * @param px value in string px value
 */
export function pxStringToNumber({ value }: { value: string }): number | undefined {
  if (!value?.length) {
    return undefined;
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return undefined;
  }
  const conversion = Number(match[0]);
  if (conversion === undefined || conversion === NaN || conversion === null) {
    return undefined;
  }

  return conversion;
}
