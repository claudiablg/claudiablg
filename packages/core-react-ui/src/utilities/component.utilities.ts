import { pascal } from 'case';
import { keys } from './utilities';

/**
 * <Link /> => "Link Medium Underline"
 */
export function getDefaultTextFromProps(name: string, props: object): string {
  const propString = `${keys(props)
    .map((prop) => pascal(props[prop]))
    .join(' ')}`;

  return `${pascal(name)} ${propString}`;
}
