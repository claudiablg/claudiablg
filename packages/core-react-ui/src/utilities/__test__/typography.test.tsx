import { defaultTypography } from '../../design-system/default-typography';
import { getCSSTypography } from '../typography.utilities';
import { defaultBaseFontSize } from '../../design-system/default-sizing';
import { createCSSCapsizeTextStyle } from '../text.utilities';
import { TextStyle, CapsizeTextStyle } from '@newrade/core-design-system';
import { defaultSansFont, defaultFallbackSansFonts } from '../../design-system/default-fonts';

describe('colors utilities', () => {
  describe(`${createCSSCapsizeTextStyle.name}`, () => {
    it('should create a valid CSS typography object', () => {
      const textStyle: TextStyle & CapsizeTextStyle = {
        font: [defaultSansFont, ...defaultFallbackSansFonts],
        capHeight: 55,
        lineGap: 19,
      };
      expect(createCSSCapsizeTextStyle({ baseFontSize: defaultBaseFontSize, ...textStyle })).toMatchSnapshot();
    });
  });

  describe(`${getCSSTypography.name}`, () => {
    it('should create a valid CSS typography object', () => {
      expect(getCSSTypography({ ...defaultTypography, baseFontSize: defaultBaseFontSize })).toMatchSnapshot();
    });
  });
});
