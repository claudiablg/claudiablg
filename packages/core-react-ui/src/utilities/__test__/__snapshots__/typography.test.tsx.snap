// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors utilities createCSSCapsizeTextStyle should create a valid CSS typography object 1`] = `
Object {
  "capHeight": 55,
  "capsize": Object {
    "::after": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginBottom": "-0.1371em",
    },
    "::before": Object {
      "content": "''",
      "display": "block",
      "height": 0,
      "marginTop": "-0.1098em",
    },
    "fontSize": "77.3626px",
    "lineHeight": "74px",
    "padding": "0.05px 0",
  },
  "font": Array [
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "link": "https://fonts.google.com/specimen/Roboto",
      "name": "Roboto",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "ui-sans-serif",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "system-ui",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "-apple-system",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "BlinkMacSystemFont",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "\\"Segoe UI\\"",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "Roboto",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "\\"Helvetica Neue\\"",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "Arial",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "\\"Noto Sans\\"",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "sans-serif",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "\\"Apple Color Emoji\\"",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "\\"Segoe UI Emoji\\"",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "\\"Segoe UI Symbol\\"",
    },
    Object {
      "fontMetrics": Object {
        "ascent": 1900,
        "capHeight": 1456,
        "descent": -500,
        "lineGap": 0,
        "unitsPerEm": 2048,
      },
      "name": "\\"Noto Color Emoji\\"",
    },
  ],
  "fontFamily": "Roboto,ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,\\"Segoe UI\\",Roboto,\\"Helvetica Neue\\",Arial,\\"Noto Sans\\",sans-serif,\\"Apple Color Emoji\\",\\"Segoe UI Emoji\\",\\"Segoe UI Symbol\\",\\"Noto Color Emoji\\"",
  "fontStyle": undefined,
  "fontWeight": undefined,
  "letterSpacing": "0px",
  "lineGap": 19,
  "textTransform": undefined,
}
`;

exports[`colors utilities getCSSTypography should create a valid CSS typography object 1`] = `
Object {
  "fonts": Object {
    "monospace": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 984,
          "capHeight": 660,
          "descent": -273,
          "lineGap": 0,
          "unitsPerEm": 1000,
        },
        "link": "https://fonts.google.com/specimen/Source+Sans+Pro",
        "name": "Source Sans Pro",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-monospace",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "SFMono-Regular",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Menlo",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Monaco",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Consolas",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Liberation Mono\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Courier New\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "monospace",
      },
    ],
    "sans": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "link": "https://fonts.google.com/specimen/Roboto",
        "name": "Roboto",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-sans-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "system-ui",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "-apple-system",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "BlinkMacSystemFont",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Segoe UI\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Roboto",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Helvetica Neue\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Arial",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Noto Sans\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "sans-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Apple Color Emoji\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Segoe UI Emoji\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Segoe UI Symbol\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Noto Color Emoji\\"",
      },
    ],
    "sansAlternate": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "link": "https://fonts.google.com/specimen/Roboto",
        "name": "Roboto",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-sans-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "system-ui",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "-apple-system",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "BlinkMacSystemFont",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Segoe UI\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Roboto",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Helvetica Neue\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Arial",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Noto Sans\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "sans-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Apple Color Emoji\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Segoe UI Emoji\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Segoe UI Symbol\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Noto Color Emoji\\"",
      },
    ],
    "serif": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 848,
          "capHeight": 660,
          "descent": -260,
          "lineGap": 0,
          "unitsPerEm": 1000,
        },
        "link": "https://fonts.google.com/specimen/Quattrocento",
        "name": "Quattrocento",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Georgia",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Cambria",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Times New Roman\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Times",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "serif",
      },
    ],
    "var": Array [
      "--font-family-monospace",
      "--font-family-sans",
      "--font-family-sans-alternate",
      "--font-family-serif",
    ],
    "varNames": Array [
      "var(--font-family-monospace)",
      "var(--font-family-sans)",
      "var(--font-family-sans-alternate)",
      "var(--font-family-serif)",
    ],
  },
  "headings": Object {
    "desktop": Object {
      "h1": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h2": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h3": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h4": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "font": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 848,
          "capHeight": 660,
          "descent": -260,
          "lineGap": 0,
          "unitsPerEm": 1000,
        },
        "link": "https://fonts.google.com/specimen/Quattrocento",
        "name": "Quattrocento",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Georgia",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Cambria",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Times New Roman\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Times",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "serif",
      },
    ],
    "mobile": Object {
      "h1": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h2": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h3": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h4": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "tablet": Object {
      "h1": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h2": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h3": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "h4": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
  },
  "labels": Object {
    "desktop": Object {
      "medium": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "small": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "xSmall": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "font": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 848,
          "capHeight": 660,
          "descent": -260,
          "lineGap": 0,
          "unitsPerEm": 1000,
        },
        "link": "https://fonts.google.com/specimen/Quattrocento",
        "name": "Quattrocento",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Georgia",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Cambria",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Times New Roman\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Times",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "serif",
      },
    ],
    "mobile": Object {
      "medium": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "small": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "xSmall": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "styles": Object {
      "bold": Object {
        "fontStyle": undefined,
        "fontWeight": 500,
        "textTransform": undefined,
      },
      "boldUppercase": Object {
        "fontStyle": undefined,
        "fontWeight": 500,
        "textTransform": "uppercase",
      },
      "uppercase": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": "uppercase",
      },
    },
    "tablet": Object {
      "medium": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "small": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "xSmall": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
  },
  "paragraphs": Object {
    "desktop": Object {
      "large": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "medium": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "small": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "xSmall": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "font": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 848,
          "capHeight": 660,
          "descent": -260,
          "lineGap": 0,
          "unitsPerEm": 1000,
        },
        "link": "https://fonts.google.com/specimen/Quattrocento",
        "name": "Quattrocento",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Georgia",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Cambria",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Times New Roman\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Times",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "serif",
      },
    ],
    "mobile": Object {
      "large": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "medium": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "small": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "xSmall": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "styles": Object {
      "bold": Object {
        "fontStyle": undefined,
        "fontWeight": 500,
        "textTransform": undefined,
      },
      "italic": Object {
        "fontStyle": "italic",
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "tablet": Object {
      "large": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "medium": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "small": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "xSmall": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
  },
  "titles": Object {
    "desktop": Object {
      "t1": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "t2": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "font": Array [
      Object {
        "fontMetrics": Object {
          "ascent": 848,
          "capHeight": 660,
          "descent": -260,
          "lineGap": 0,
          "unitsPerEm": 1000,
        },
        "link": "https://fonts.google.com/specimen/Quattrocento",
        "name": "Quattrocento",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "ui-serif",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Georgia",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Cambria",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "\\"Times New Roman\\"",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "Times",
      },
      Object {
        "fontMetrics": Object {
          "ascent": 1900,
          "capHeight": 1456,
          "descent": -500,
          "lineGap": 0,
          "unitsPerEm": 2048,
        },
        "name": "serif",
      },
    ],
    "mobile": Object {
      "t1": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "t2": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
    "tablet": Object {
      "t1": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
      "t2": Object {
        "fontStyle": undefined,
        "fontWeight": 400,
        "textTransform": undefined,
      },
    },
  },
}
`;
